retro_data_structures/formats/script_layer.py

```diff
@@ -210,3 +212,20 @@ def is_modified(self) -> bool:
 
     def mark_modified(self):
         self._modified = True
+    
+    def build_mlvl_dependencies(self, asset_manager: AssetManager) -> typing.Iterator[Dependency]:
+        deps: list[Dependency] = []
+        for instance in self.instances:
+            deps.extend(instance.mlvl_dependencies_for(asset_manager))
+        
+        # only unique dependencies
+        yield from list(dict.fromkeys(deps))
+
+    @property
+    def dependencies(self):
+        self.assert_parent()
+        deps = self._parent_area._raw.dependencies
+        offsets = deps.offsets
+        raw_deps = deps.dependencies[offsets[self._index]:offsets[self._index+1]]
+        yield from [(dep.asset_type, dep.asset_id) for dep in raw_deps]
```

Comment:
Replace `[]` with `()` so we don't make a pointless list.

pretty minor thing though :p